Refactor queues

bebehei opened the pull request:
Ok, I'm done with the queue refactoring.

- Every part about accessing a `GQueue` is now handled in `queues.c`.
- All methods handling a queue are prefixed with `queues_` to resemble the namespace.
- The three queues have unique names `waiting`, `displayed`, `history`.
- All stuff in `notification_init` about closing notifications is removed
    - to `dbus.c`: command pause/resume interpretation
    - to `queues.c`: duplicates stacking

The refactoring about splitting `notification_init` is still to be done, but not part of this PR.

Mostly, I have split the stuff into two commits (the first one plain moving code from `file.c` to `queues.c` and the second one renaming/refactoring).

All commits compile without warnings.

Albeit most of the commits are plain refactoring, two commits stand out and may need further testing:

- `Force management of queues to queues.c`
- `Remove command handling from notification_init `
- `Move id assignment to separate function in queues.c`

tsipinakis commented on src/notification.c:
```diff
@@ -296,7 +287,16 @@ int notification_init(notification *n, int id)
+        if(n->category == NULL) n->category = g_strdup("");
+        n->redisplayed = false;
+        n->first_render = true;
+        n->script = NULL;
```
Nitpick but not sure these are needed, notifications should be created via `create_notification` which returns zero-ed out memory anyway.

bebehei replied:
Yeah that stuff about `notification_init`. I only moved these values to the top of the notification to show, that these are actually not really processed in `notification_init`.

bebehei commented:
> Haven't yet sat down to do a detailed review of this, but here are some passing remarks after messing around for a few minutes:

Yes, I know. I've currently experiencing some weird segfaults (which are actually outside of this PR).

Please wait with a detailed review. I will ping you then again.

bebehei commented on src/queues.c:
```diff
@@ -213,6 +213,37 @@ void history_push(notification *n)
                 g_queue_push_tail(history, n);
 }
 
+gint64 queues_get_next_datachange(gint64 time)
```
Yes, there seems to be no rationale for this. I don't know why I thought this would be neccessary. Maybe because I have to pass the output of `x_is_idle`.

bebehei commented:
From IRC:

> Is #411 supposed to be ready for a review? You said you'd ping me but never did.

@tsipinakis I've written you a few days ago, that it's working fine and I have not found a bug for the last week. I never said it explicitly, but yes it was this type of "ping".

bebehei commented:
I found another memory issue (which also is on current master): Notifications marked with `history_ignore` never get freed. This is fixed in the last commit now.

bebehei commented:
@tsipinakis 

Thread Bene:
pr.lock()
write_tests()

I started writing tests for this. But: first I refactored `notification_init`, so that I can actually test the queues stuff. And I rebased the whole thing on current master, because merge conflicts with the #418 PR came along.

tsipinakis commented:
From IRC

> 20:09:49 bebehei | Hey nikos, I've chopped off the commits from 411 I added yesterday. I will add the tests in a different PR.


Though it looks like you also removed some bugfix commits like the notification not being freed if it isn't pushed to the history or the timeout fix as mentioned in #419

bebehei commented:
Yes, I chopped off all unrelated commits. The other stuff will get merged via different PR.
